table_schema_cache.go

```diff
@@ -293,6 +293,31 @@ func (c TableSchemaCache) Get(database, table string) *TableSchema {
 	return c[fullTableName(database, table)]
 }
 
+// Helper to sort a given map of tables with a second list giving a priority.
+// If an element is present in the input and the priority lists, the item will
+// appear first (in the order of the priority list), all other items appear in
+// the order given in the input
+func (c TableSchemaCache) GetTableListWithPriority(priorityList []string) (tableNames []string) {
```

Comment:
> I would remove the param from this method and make it implicitly use the configured TableCreationOrder as it doesn't appear to be necessary.

the method is defined on `TableSchemaCache` which is defined as

```
type TableSchemaCache map[string]*TableSchema
```

as such, it is self-contained and does not have access to the config. I think passing it in is required - unless I'm misunderstanding your suggestion

Reply:
Yes, sorry, I got ahead of myself and didn't fully explain. 

This would require moving the `TableCreationOrder` to the base configuration struct as well as the `TableSchema` struct instead of in `copydb/config.go` (similar to the `CompressedColumnsForVerification`). It can then be passed to the `LoadTables` func, although that method's params do seem to be getting a bit unwieldy. 

If you're in on board, it may be easier at this point to add an additional struct to be used simply for the `LoadTables` function - something like a `TableConfiguration` that contains all of the params as struct members.

Reply:
I'm not against it, but I'd like to hear other opinions. I personally feel it's more of a change than necessary for this specific change. I'm a strong proponent for minimal patch-sets whenever possible (as you may have seen from the 1734 issues I have opened and sent separate PRs for ;-) .

I **do** agree, however, that the number of parameters passed to `LoadTables` is insane :-) and a dedicated struct seems very reasonable. But I would say that it's orthogonal to the proposed change here

Reply:
That's fair. I don't feel strongly enough to block the PR, but was simply proposing an alternative that I felt would make more sense with a bit larger change.